### DISCUSSION TITLE
🐞 OCI labels no longer work by default in ghcr.io since v0.6.3

### DISCUSSION THREAD

**Discussion by gerhard** [dagger/dagger#12475] [Help]
### What is the issue?

When `MIMEType` defaults to `application/vnd.oci.image.layer.v1.tar+gzip`, ghcr.io no longer displays OCI labels correctly, i.e. https://github.com/users/gerhard/packages/container/sysadmin/114534202

This changed in [`v0.6.3`](https://github.com/dagger/dagger/releases/tag/v0.6.3) via:
- https://github.com/dagger/dagger/pull/5365

---

When `MIMEType` is set to `application/vnd.oci.image.layer.v1.tar.gzip`, OCI labels work  as expected, i.e. https://github.com/users/gerhard/packages/container/sysadmin/114536750

This is achieved by explicitly setting `MediaTypes` to `dagger.Dockermediatypes` in the context of `Publish()`, e.g.:

```go
container.
	Pipeline("publish").
	WithRegistryAuth("ghcr.io", registryUsername, registryPassword).
	WithLabel(
		"org.opencontainers.image.description",
		"🇨🇭 A Swiss army knife for all things sysadmin").
	Publish(ctx, imageURL, dagger.ContainerPublishOpts{
		MediaTypes: dagger.Dockermediatypes,
	})
```

This was made possible in [`v0.6.3`](https://github.com/dagger/dagger/releases/tag/v0.6.4) via:
- https://github.com/dagger/dagger/pull/5467

---

I am wondering if we should default to `dagger.Dockermediatypes` since that seems to be preferred ghcr.io format.

WDYT @sipsma @vito @aluzzardi?

---

> **Note**
> `skopeo inspect docker://ghcr.io/gerhard/sysadmin@sha256:1def2d31726ed5be8761d5eb1da5ded5ce3f2fcbf7a62e302b2013860d8b7913`

### SDK version

Go SDK v0.7.3 & v0.7.4

### OS version

SDK / CLI on macOS 12.6 & Engine on NixOS 22.11 (Linux 5.15)

**Comment by sipsma** [comment]
GHCR docs claim to support oci image spec annotations: https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-container-registry#labelling-container-images

So if I'm reading them right and they are to be believed, I think something else is probably breaking.

I know we have integ tests for publishing all the different mediatypes, and I think we have integ tests for labels, so it may be worth creating a mashup test and see what happens.

**Comment by gerhard** [comment]
https://github.com/users/gerhard/packages/container/sysadmin/114534202 does not have the `labels` key at the bottom of the manifest:

<img width="1440" alt="image" src="https://github.com/dagger/dagger/assets/3342/12508dad-783b-40b2-95c2-8e466969e515">

https://github.com/users/gerhard/packages/container/sysadmin/114536750 on the other hand does:

<img width="1440" alt="image" src="https://github.com/dagger/dagger/assets/3342/9510ef37-a202-4535-a58d-c9fa2cf1df07">

Will add an integration test & continue digging further. Thank you @sipsma!

**Comment by nipuna-perera** [comment]
Hey @gerhard I had the same issue with an older version of Artifactory that doesn't support OCI images and using `DockerMediaTypes` when publishing worked for me. Ref #5433

**Comment by gerhard** [comment]
In this case, publishing works, it's just the labels that don't appear correctly in the registry. I have confirmed that the local manifest is correct, when using `Export()`, it must be `Publish()` related and/or the registry itself. Will keep digging, thanks!
